tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -11330,6 +11343,124 @@ TEST_F(ExecutionTest, QuadAnyAll) {
     WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
 }
 
+TEST_F(ExecutionTest, IsNormalTest) {
+  EnableShaderBasedValidation();
+    WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  CComPtr<IStream> pStream;
+  ReadHlslDataIntoNewStream(L"ShaderOpArith.xml", &pStream);
+
+  std::shared_ptr<st::ShaderOpSet> ShaderOpSet =
+      std::make_shared<st::ShaderOpSet>();
+  st::ParseShaderOpSetFromStream(pStream, ShaderOpSet.get());
+  st::ShaderOp *pShaderOp =
+      ShaderOpSet->GetShaderOp("IsNormal");
+  vector<st::ShaderOpRootValue> fallbackRootValues = pShaderOp->RootValues;
+
+  const int expectedResultsSize = 12;
+  // false represents 0, true represents a non-zero value
+  int expectedResults[expectedResultsSize] = {0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1};
+    
+  // TestShaderModels will be an array, where the first x models are "non-fallback", and the rest of the models
+  // are "fallback". If TestShaderModels has length y, and a test loops through all shader models, a convention
+  // to test based on whether fallback is enabled or not is to limit the loop like this:
+  // unsigned num_models_to_test = ExecutionTest::IsFallbackPathEnabled() ? y : x;
+  D3D_SHADER_MODEL sm = D3D_SHADER_MODEL_6_0;
+  LogCommentFmt(L"\r\nVerifying isNormal in shader "
+                L"model 6.%1u",
+                ((UINT)sm & 0x0f));
+
+  CComPtr<ID3D12Device> pDevice;
+  CreateDevice(&pDevice, sm, false /* skipUnsupported */);
+  D3D12_FEATURE_DATA_D3D12_OPTIONS devOptions;
+  VERIFY_SUCCEEDED(
+      pDevice->CheckFeatureSupport((D3D12_FEATURE)D3D12_FEATURE_D3D12_OPTIONS,
+                                    &devOptions, sizeof(devOptions)));
+  if (devOptions.ResourceBindingTier < D3D12_RESOURCE_BINDING_TIER_3) {
+    WEX::Logging::Log::Comment(
+        L"Device does not support Resource Binding Tier 3");
+    WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+    return;
+  }
+
+  auto ShaderInitFn = [&](LPCSTR Name, LPCSTR pText, IDxcBlob **ppShaderBlob, st::ShaderOp *pShaderOp) {
+    UNREFERENCED_PARAMETER(pShaderOp);
+    UNREFERENCED_PARAMETER(Name);
+    CComPtr<IDxcUtils> pUtils;
+    VERIFY_SUCCEEDED(m_support.CreateInstance(CLSID_DxcUtils, &pUtils));
+    // Compile original HLSL with op to replace, and disassemble.
+    CComPtr<IDxcBlob> compiledShader;
+    std::vector<LPCWSTR> args; // Add args here
+    VerifyCompileOK(m_support, pText, L"cs_6_0", args, &compiledShader);
+    std::string disassembly = DisassembleProgram(m_support, compiledShader);
+    // Replace op
+    ReplaceText(
+      { "@dx.op.isSpecialFloat.f32(i32 8," },
+      { "@dx.op.isSpecialFloat.f32(i32 11," },
```

Comment:
This function only replaces the f32 operation.  Double will require another replacement, like so:
```
      { "@dx.op.isSpecialFloat.f32(i32 8,",  "@dx.op.isSpecialFloat.f64(i32 8," },
      { "@dx.op.isSpecialFloat.f32(i32 11,", "@dx.op.isSpecialFloat.f64(i32 11," },
```